src/Heist/Extra/Splices/Pandoc/Render.hs

```diff
@@ -224,10 +224,16 @@ rpInline' ctx@RenderCtx {..} i = case i of
             <> rpAttr (concatAttr attr $ iAttr i)
     one . X.Element "a" attrs <$> foldMapM (rpInline ctx) is
   B.Image attr is (url, tit) -> do
-    let attrs =
-          catMaybes [pure ("src", url), guard (not $ T.null tit) >> pure ("title", tit), pure ("alt", WL.plainify is)]
-            <> rpAttr (rewriteClass ctx attr)
-    pure $ one . X.Element "img" attrs $ mempty
+    let lowerUrl = T.toLower url
+    if T.isInfixOf ".pdf?t=" lowerUrl || T.isSuffixOf ".pdf" lowerUrl
+      then do
+        tpl <- HE.lookupHtmlTemplateMust "/templates/filters/embed-pdf"
+        HE.runCustomTemplate tpl ("ema:url" ## HI.textSplice url)
+      else do
+        let attrs =
+              catMaybes [pure ("src", url), guard (not $ T.null tit) >> pure ("title", tit), pure ("alt", WL.plainify is)]
+                <> rpAttr (rewriteClass ctx attr)
+        pure $ one . X.Element "img" attrs $ mempty
```

Comment:
`src/Heist/Extra` is to be considered a general library; it shouldn't be hardcoded with Emanote-specific stuff (such as the templates).

What is this code doing here anyway? When I revert this change, the PDF demo still works.

Reply:
This code is required for markdown links `[book] (./book.pdf)`

Reply:
Why would we want to embed regular markdown links?

Reply:
Without this code, pdf is rendered as `<img>` tag which is not working

Reply:
Is linking to pdf using image syntax a common practice? What other editors (Obsidian?) support this non standard practice?

Reply:
@srid yes, Obsidian uses this syntax. I didn't use other PKM but I think such syntax the most likely is used for pdfs heavily

Reply:
@srid Well, ideally the underline parser distinguishes between `Image` and `Pdf` token, but I as it is not the case now, I consider what I wrote as a hack

Reply:
> I think the implementation should then be refactored.

@mnaoumov If you'd like to take a stab at doing this, check out the `InlineRef` type (we could add an `InlinePDF` constructor in addition to `InlineImage`), as well as the `urlResolvingSplice` function.

https://github.com/EmaApps/emanote/blob/6433224d6a23577995e9c14fa3bdd75bddf815d8/src/Emanote/Pandoc/Link.hs#L10-L22

https://github.com/EmaApps/emanote/blob/6433224d6a23577995e9c14fa3bdd75bddf815d8/src/Emanote/Pandoc/Renderer/Url.hs#L42-L45

Reply:
I'll merge this PR, but without the regular image linking support because it will take some time to refactor Emanote to do this properly. It can happen in separate PR.